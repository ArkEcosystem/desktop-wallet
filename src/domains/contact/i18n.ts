export const translations: { [key: string]: any } = {
	CONTACTS_PAGE: {
		ADD_CONTACT: "Add Contact",
		ADD_CONTACT_MESSAGE: "Add your most frequent contacts for fast, easy payments",
		TITLE: "My Contacts",
		SUBTITLE: "Manage your frequent contacts",
	},
	CONTACT_FORM: {
		ADD_ADDRESS: "Add address",
		ADDRESS: "Address",
		ADDRESSES: "Addresses",
		DELETE_CONTACT: "Delete Contact",
		NAME: "Name",
		NETWORK: "Network",
	},
	MODAL_CREATE_CONTACT: {
		TITLE: "Add Contact",
		DESCRIPTION: "Enter your contacts name and addresses and add them to your list.",
	},
	MODAL_DELETE_CONTACT: {
		TITLE: "Delete Contact",
		DESCRIPTION: "Do you really want to delete this contact? Once deleted, you will not be able to restore it.",
	},
	MODAL_UPDATE_CONTACT: {
		TITLE: "Edit Contact",
	},
	MODAL_SEARCH_CONTACT: {
		TITLE: "Search Contact",
		DESCRIPTION: "Find the contact you need quickly and conveniently",
		PLACEHOLDER: "Search...",
	},
	VALIDATION: {
		MAXLENGTH_ERROR: "The Contact Name should have less than {{maxLength}} characters.",
		CONTACT_NAME_EXISTS: "Contact with name {{name}} already exists.",
		CONTACT_NAME_EXISTS_SUFFIX: "already exists",
	},
};
