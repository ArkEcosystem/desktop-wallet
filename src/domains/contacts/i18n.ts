export const translations: { [key: string]: any } = {
	CONTACT_FORM: {
		ADD_ADDRESS: "Add Address",
		ADDRESS: "Address",
		ADDRESSES: "Addresses",
		NAME: "Name",
		NETWORK: "Network",
	},
	MODAL_CREATE_CONTACT: {
		TITLE: "Add Contact",
		DESCRIPTION: "Enter your contacts name and addresses and add them to your list.",
	},
	MODAL_DELETE_CONTACT: {
		TITLE: "Delete Contact",
		DESCRIPTION: "Do you really want to delete this contact? Once deleted, you will not be able to restore it.",
	},
	MODAL_UPDATE_CONTACT: {
		TITLE: "Edit Contact",
	},
};
