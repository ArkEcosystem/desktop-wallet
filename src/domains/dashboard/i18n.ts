export const translations: { [key: string]: any } = {
	FILTER_WALLETS: {
		NETWORK: {
			TITLE: "Filter Network",
			DESCRIPTION: "Select the types of networks",
		},
		WALLETS: {
			TITLE: "Your Wallets",
			DESCRIPTION: "Select the display of your wallets",
		},
		CHART: {
			TITLE: "Portfolio Chart",
			DESCRIPTION: "You can hide this unit if you like",
		},
		TRANSACTION_HISTORY: {
			TITLE: "Transaction History",
			DESCRIPTION: "You can hide this unit if you like",
		},
	},

	TRANSACTION_HISTORY: {
		TITLE: "Transactions History",
		EMPTY_TEXT:
			"This will display the history of your transactions. But you don't have more than one transaction at the moment.",
	},

	WALLET_CONTROLS: {
		CREATE: "Create",
		IMPORT: "Import",
	},

	DASHBOARD_PAGE: {
		CHART: {
			TITLE: "Portfolio Chart",
			PERCENTAGES_LABEL: "Total portfolio",
		},
	},
};
