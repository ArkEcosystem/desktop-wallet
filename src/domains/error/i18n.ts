export const translations: { [key: string]: any } = {
	APPLICATION: {
		TITLE: "Something went wrong",
		DESCRIPTION: "It looks like something went wrong. Try visiting another page or reloading.",
		RELOAD: "Reload",
	},
	OFFLINE: {
		TITLE: "Cannot Connect",
		DESCRIPTION: "Check your internet connection if it doesn't work, check it again.",
	},
};
