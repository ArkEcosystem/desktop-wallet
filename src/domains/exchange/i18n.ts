export const translations: { [key: string]: any } = {
	MODAL_ADD_EXCHANGE: {
		TITLE: "Exchange",
		DESCRIPTION:
			"Buy ARK directly from the ARK Desktop Wallet or through one of 20+ exchanges supplying ARK tokens using BTC, ETH or other trading pairs.",
	},
};
