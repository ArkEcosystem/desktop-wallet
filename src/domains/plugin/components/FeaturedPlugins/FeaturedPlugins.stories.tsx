import { boolean } from "@storybook/addon-knobs";
import React from "react";

import { FeaturedPlugins } from "./FeaturedPlugins";

export default { title: "Domains / Plugin / Components / FeaturedPlugins" };

export const Default = () => <FeaturedPlugins isOpen={boolean("Is Open", true)} onClose={() => alert("closed")} />;
