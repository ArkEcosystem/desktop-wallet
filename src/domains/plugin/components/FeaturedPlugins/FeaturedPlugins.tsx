import { images } from "app/assets/images";
import { Button } from "app/components/Button";
import { Icon } from "app/components/Icon";
import { Image } from "app/components/Image";
import { Modal } from "app/components/Modal";
import { ReviewRating } from "app/components/ReviewRating";
import { Table } from "app/components/Table";
import React from "react";
import { useTranslation } from "react-i18next";

type FeaturedPluginsProps = {
	isOpen: boolean;
	plugins: any;
	onClose?: any;
};

const { BestPluginsBanner } = images.plugin.common;

export const FeaturedPlugins = ({ isOpen, plugins, onClose }: FeaturedPluginsProps) => {
	const { t } = useTranslation();

	const columns = [
		{
			Header: " ",
			disableSortBy: true,
		},
		{
			Header: t("COMMON.NAME"),
			accessor: "name",
		},
		{
			Header: t("COMMON.CATEGORY"),
			accessor: "category",
			className: "justify-center",
		},
		{
			Header: t("COMMON.RATING"),
			accessor: "rating",
			className: "justify-center",
		},
		{
			Header: t("COMMON.VERSION"),
			accessor: "version",
			className: "justify-center",
		},
		{
			Header: "  ",
			disableSortBy: true,
		},
	];

	return (
		<Modal
			title={t("PLUGINS.MODAL_FEATURED_PLUGINS.TITLE")}
			description={t("PLUGINS.MODAL_FEATURED_PLUGINS.DESCRIPTION")}
			banner={<BestPluginsBanner className="h-full" />}
			size="4xl"
			isOpen={isOpen}
			onClose={onClose}
		>
			<div className="mt-8 -mb-6">
				<Table columns={columns} data={plugins}>
					{(rowData: any) => (
						<tr className="border-b border-dashed border-theme-neutral-300 dark:border-theme-neutral-800">
							<td className="w-16">
								<Image name="ChangeNowLogo" domain="exchange" className="w-12 h-12" />
							</td>

							<td>
								<div className="font-semibold text-theme-primary-500 hover:text-theme-primary-400">
									{rowData.name}
								</div>
								<div className="inline-flex items-center space-x-2">
									<span className="text-theme-secondary-text">{rowData.author}</span>
									{rowData.isOfficial && <Icon name="OfficialArkPlugin" width={15} height={15} />}
									{rowData.isGrant && <Icon name="Grant" width={16} height={16} />}
								</div>
							</td>

							<td className="py-10 text-center text-theme-secondary-text">
								{t(`PLUGINS.CATEGORIES.${rowData.category.toUpperCase()}`)}
							</td>

							<td className="flex justify-center py-10 text-theme-secondary-text">
								<ReviewRating rating={rowData.rating} width={3} />
							</td>

							<td className="py-10 text-center text-theme-secondary-text">v {rowData.version}</td>

							<td className="w-16">
								<Button variant="secondary">{t("COMMON.INSTALL")}</Button>
							</td>
						</tr>
					)}
				</Table>
			</div>
		</Modal>
	);
};

FeaturedPlugins.defaultProps = {
	isOpen: false,
};
