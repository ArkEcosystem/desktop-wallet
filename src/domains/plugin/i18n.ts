export const translations: { [key: string]: any } = {
	NEW_UPDATES_AVAILABLE: "New updates available",
	NEW_VERSION_AVAILABLE: "New version available",

	UPDATE_ALL_NOTICE: "There is {{count}} update available for your plugins. Would you like to update it?",
	UPDATE_ALL_NOTICE_plural:
		"There are {{count}} updates available for your plugins. Would you like to update them all now?",
	UPDATE_ALL: "Update All",

	MINIMUM_VERSION_NOT_SATISFIED:
		"In order to update this plugin please update Desktop Wallet to v{{minimumVersion}}+",

	WARNING_DISCLAIMER:
		"Please make sure to check the documentation of the plugin before installing it. By installing it on your wallet, you assume every responsibility.",

	DEVELOPER_RESPONSE: "Developer response",

	PERMISSIONS: {
		EVENTS: "Allows access to the Desktop Wallet events",
		FILESYSTEM: "Allows using file dialogs",
		HTTP: "Allows performing external web requests",
		LAUNCH: "Allows to register a custom view",
		PROFILE: "Allows access to the currently active profile",
		STORE: "Allows storing data within the Desktop Wallet, using key-value pairs",
		THEME: "Allows loading additional custom themes for the Desktop Wallet",
		TIMERS: "Allows using timeouts and intervals",

		// Legacy permissions
		ALERTS: "Allows access to the Desktop Wallet alerts",
		AUDIO: "Allows access to play audio from within the Desktop Wallet",
		AVATARS: "Plugin contains custom avatars",
		COMPONENTS: "Allows loading custom components",
		DIALOGS: "Allows using file dialogs",
		MENU_ITEMS: "Allows adding custom menu items to the Desktop Wallet sidebar",
		MESSAGING: "Allows WebFrames access to a one-way messaging system",
		PEER_ALL: "Allows access to the peer discovery",
		PEER_CURRENT: "Allows access to the currently connected peer",
		PUBLIC: "Allows navigation to wallet routes and provides access to the Font Awesome icon set",
		PROFILE_ALL: "Allows access to all available profiles",
		PROFILE_CURRENT: "Allows access to the currently active profile",
		ROUTES: "Allows loading additional routes into the Desktop Wallet",
		STORAGE: "Allows storing data within the Desktop Wallet, using key-value pairs",
		THEMES: "Allows loading additional custom themes for the Desktop Wallet",
		LANGUAGES: "Allows loading additional languages for the Desktop Wallet",
		UI_COMPONENTS: "Allows access to the standard Desktop Wallet components used throughout",
		UTILS: "Allows using utilities such as the BigNumber type and dayjs",
		WALLET_TABS: "Allows showing an additional tab/page on the Wallet screen",
		WEBFRAME: "Allows showing remote URL pages within a frame",
		WEBSOCKET: "Allows connections to websockets",
	},

	MODAL_UNINSTALL: {
		TITLE: "Uninstall Plugin",
		DESCRIPTION:
			"Are you sure you want to remove '{{ name }}'? Uninstalling this plugin will remove it from all profiles on this system.",
	},

	MODAL_INSTALL_PLUGIN: {
		DESCRIPTION: "This plugin needs the following permissions:",
		SUCCESS: `The plugin "{{name}}" was successfully installed`,
		INSTALL_FAILURE: `Failed to install plugin "{{name}}"`,
		DOWNLOAD_FAILURE: `Failed to donwload plugin "{{name}}"`,
	},

	MODAL_MANUAL_INSTALL_PLUGIN: {
		TITLE: "Install from URL",
		DESCRIPTION:
			"Fetch the plugin directly from GitHub by using the URL of the plugin repository in the input field below.",
		PLACEHOLDER: "Github Repository URL",
	},

	PAGE_PLUGINS_CATEGORY: {
		LAYOUT_TITLE: "Plugin list",
	},

	PAGE_PLUGIN_MANAGER: {
		TITLE: "Plugin Manager",
		DESCRIPTION: "Easy way to find, manage and install plugins",

		NO_PLUGINS_AVAILABLE: "This category has no available plugins yet.",

		VIEW: {
			HOME: "Home",
			GAMING: "Gaming",
			MY_PLUGINS: "My Plugins",
			OTHER: "Other",
			EXCHANGE: "Exchange",
			UTILITY: "Utility",
		},
	},

	PLUGIN_INFO: {
		ABOUT: "About the plugin",
		PERMISSIONS: "Permissions",
		SCREENSHOTS: "Screenshots",
		REQUIREMENTS: "Requirements",
		DESKTOP_WALLET_VERSION: "Desktop Wallet Version",
	},

	CATEGORIES: {
		GAMING: "Gaming",
		GAME: "Game",
		LANGUAGE: "Language",
		MY_PLUGINS: "My Plugins",
		OTHER: "Other",
		UTILITY: "Utility",
		EXCHANGE: "Exchange",
	},

	FILTERS: {
		CATEGORIES: {
			GAME: "Game",
			UTILITY: "Utility",
			EXCHANGE: "Exchange",
			OTHER: "Other",
		},
	},
};
