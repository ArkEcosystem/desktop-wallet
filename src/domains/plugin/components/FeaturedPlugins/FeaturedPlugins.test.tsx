import { render } from "@testing-library/react";
import { i18n } from "app/i18n";
import React from "react";
import { I18nextProvider } from "react-i18next";

import { translations } from "../../i18n";
import { FeaturedPlugins } from "./FeaturedPlugins";

describe("FeaturedPlugins", () => {
	it("should not render if not open", () => {
		const { asFragment, getByTestId } = render(
			<I18nextProvider i18n={i18n}>
				<FeaturedPlugins isOpen={false} />
			</I18nextProvider>,
		);

		expect(() => getByTestId("modal__inner")).toThrow(/Unable to find an element by/);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render a modal", () => {
		const { asFragment, getByTestId } = render(
			<I18nextProvider i18n={i18n}>
				<FeaturedPlugins isOpen={true} />
			</I18nextProvider>,
		);

		expect(getByTestId("modal__inner")).toHaveTextContent(translations.MODAL_FEATURED_PLUGINS.TITLE);
		expect(getByTestId("modal__inner")).toHaveTextContent(translations.MODAL_FEATURED_PLUGINS.DESCRIPTION);
		expect(asFragment()).toMatchSnapshot();
	});
});
