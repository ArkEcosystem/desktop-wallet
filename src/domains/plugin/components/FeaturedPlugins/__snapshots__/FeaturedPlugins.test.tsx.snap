// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedPlugins should not render if not open 1`] = `<DocumentFragment />`;

exports[`FeaturedPlugins should render a modal 1`] = `
<DocumentFragment>
  <div
    class="fixed inset-0 z-50 bg-black opacity-50"
    data-testid="modal__overlay"
  />
  <div
    class="sc-AxiKw jCpCzv fixed top-0 left-0 right-0 z-50 flex flex-col px-10 pt-6 pb-8 mx-auto mt-24 overflow-hidden rounded-xl bg-theme-background"
    data-testid="modal__inner"
  >
    <div
      class="absolute top-0 right-0 z-50 mt-5 mr-5"
    >
      <button
        class="sc-AxjAm bThAoP"
        color="neutral"
        data-testid="modal__close-btn"
        type="button"
      >
        <div
          class="p-1"
        >
          <div
            class="sc-AxirZ hDwaov"
            height="12"
            width="12"
          >
            <svg>
              cross-slim.svg
            </svg>
          </div>
        </div>
      </button>
    </div>
    <div
      class="py-4"
    >
      <div
        class="relative mb-10 -mx-10 -mt-10"
      >
        <svg
          class="w-full"
        >
          best-plugins-banner.svg
        </svg>
        <h1
          class="absolute bottom-0 left-0 mb-8 ml-12"
        >
          <div
            class="mb-5 text-5xl font-bold leading-tight text-white"
          >
            Featured Plugins
          </div>
        </h1>
      </div>
      <div
        class="flex-1"
      >
        <div
          class="mt-1 text-theme-neutral-dark"
        >
          ARK is an open-source blockchain ecosystem that provides users with innovate and easy-to-use blockchain thechnologies. Our supportive community, extensive range of products and easy-to-implement technology, underpinned by the ARK Public Network, empowers individuals to adopt and apply blockhain tecnology in their everyday lives.
        </div>
        <div
          class="mt-8 -mb-6"
        >
          <div
            class="sc-AxhCb tChBQ"
            role="table"
          >
            <table
              class="table-auto"
            >
              <thead>
                <tr
                  role="row"
                >
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--0"
                    role="columnheader"
                  >
                    <div
                      class="flex flex-inline align-top undefined"
                    >
                      <div>
                         
                      </div>
                    </div>
                  </th>
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--1"
                    role="columnheader"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    <div
                      class="flex flex-inline align-top undefined"
                    >
                      <div>
                        Name
                      </div>
                      <div
                        class="flex items-center ml-2 text-theme-color-neutral"
                        data-testid="table__Sort"
                      >
                        <div
                          class="sc-AxirZ jMwzyA"
                          height="12"
                          width="15"
                        >
                          <svg>
                            sort.svg
                          </svg>
                        </div>
                      </div>
                    </div>
                  </th>
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--2"
                    role="columnheader"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    <div
                      class="flex flex-inline align-top justify-center"
                    >
                      <div>
                        Category
                      </div>
                      <div
                        class="flex items-center ml-2 text-theme-color-neutral"
                        data-testid="table__Sort"
                      >
                        <div
                          class="sc-AxirZ jMwzyA"
                          height="12"
                          width="15"
                        >
                          <svg>
                            sort.svg
                          </svg>
                        </div>
                      </div>
                    </div>
                  </th>
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--3"
                    role="columnheader"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    <div
                      class="flex flex-inline align-top justify-center"
                    >
                      <div>
                        Rating
                      </div>
                      <div
                        class="flex items-center ml-2 text-theme-color-neutral"
                        data-testid="table__Sort"
                      >
                        <div
                          class="sc-AxirZ jMwzyA"
                          height="12"
                          width="15"
                        >
                          <svg>
                            sort.svg
                          </svg>
                        </div>
                      </div>
                    </div>
                  </th>
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--4"
                    role="columnheader"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    <div
                      class="flex flex-inline align-top justify-center"
                    >
                      <div>
                        Version
                      </div>
                      <div
                        class="flex items-center ml-2 text-theme-color-neutral"
                        data-testid="table__Sort"
                      >
                        <div
                          class="sc-AxirZ jMwzyA"
                          height="12"
                          width="15"
                        >
                          <svg>
                            sort.svg
                          </svg>
                        </div>
                      </div>
                    </div>
                  </th>
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--5"
                    role="columnheader"
                  >
                    <div
                      class="flex flex-inline align-top undefined"
                    >
                      <div>
                          
                      </div>
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody
                role="rowgroup"
              >
                <tr
                  class="border-b border-dashed border-theme-neutral-200"
                >
                  <td
                    class="w-16"
                  >
                    <svg
                      class="w-12 h-12"
                    >
                      changenow-logo.svg
                    </svg>
                  </td>
                  <td>
                    <div
                      class="font-semibold text-theme-primary-500 hover:text-theme-primary-400"
                    >
                      ARK Explorer
                    </div>
                    <div
                      class="inline-flex items-center space-x-2"
                    >
                      <span
                        class="text-theme-neutral-dark"
                      >
                        ARK Ecosystem
                      </span>
                      <div
                        class="sc-AxirZ fkjLBb"
                        height="15"
                        width="15"
                      >
                        <svg>
                          official-ark-plugin.svg
                        </svg>
                      </div>
                    </div>
                  </td>
                  <td
                    class="py-10 text-center text-theme-neutral-dark"
                  >
                    Utility
                  </td>
                  <td
                    class="flex justify-center py-10 text-theme-neutral-dark"
                  >
                    <div
                      class="flex"
                      data-testid="ReviewRating"
                    >
                      <div
                        class="relative my-auto"
                      >
                        <div
                          class="absolute overflow-hidden"
                          style="width: 89.99999999999999%;"
                        >
                          <svg
                            class="text-theme-warning-400 w-3"
                          >
                            star.svg
                          </svg>
                        </div>
                        <svg
                          class="text-theme-neutral w-3"
                        >
                          star.svg
                        </svg>
                      </div>
                      <span
                        class="leading-tight my-auto ml-1 {{ $ratingClass ?? '' }}"
                      >
                        4.6
                      </span>
                    </div>
                  </td>
                  <td
                    class="py-10 text-center text-theme-neutral-dark"
                  >
                    v 1.3.8
                  </td>
                  <td
                    class="w-16"
                  >
                    <button
                      class="sc-AxjAm ddJBbC"
                      color="primary"
                      type="button"
                    >
                      Install
                    </button>
                  </td>
                </tr>
                <tr
                  class="border-b border-dashed border-theme-neutral-200"
                >
                  <td
                    class="w-16"
                  >
                    <svg
                      class="w-12 h-12"
                    >
                      changenow-logo.svg
                    </svg>
                  </td>
                  <td>
                    <div
                      class="font-semibold text-theme-primary-500 hover:text-theme-primary-400"
                    >
                      Animal Avatars
                    </div>
                    <div
                      class="inline-flex items-center space-x-2"
                    >
                      <span
                        class="text-theme-neutral-dark"
                      >
                        Breno Polanski
                      </span>
                    </div>
                  </td>
                  <td
                    class="py-10 text-center text-theme-neutral-dark"
                  >
                    Utility
                  </td>
                  <td
                    class="flex justify-center py-10 text-theme-neutral-dark"
                  >
                    <div
                      class="flex"
                      data-testid="ReviewRating"
                    >
                      <div
                        class="relative my-auto"
                      >
                        <div
                          class="absolute overflow-hidden"
                          style="width: 89.99999999999999%;"
                        >
                          <svg
                            class="text-theme-warning-400 w-3"
                          >
                            star.svg
                          </svg>
                        </div>
                        <svg
                          class="text-theme-neutral w-3"
                        >
                          star.svg
                        </svg>
                      </div>
                      <span
                        class="leading-tight my-auto ml-1 {{ $ratingClass ?? '' }}"
                      >
                        4.6
                      </span>
                    </div>
                  </td>
                  <td
                    class="py-10 text-center text-theme-neutral-dark"
                  >
                    v 1.3.8
                  </td>
                  <td
                    class="w-16"
                  >
                    <button
                      class="sc-AxjAm ddJBbC"
                      color="primary"
                      type="button"
                    >
                      Install
                    </button>
                  </td>
                </tr>
                <tr
                  class="border-b border-dashed border-theme-neutral-200"
                >
                  <td
                    class="w-16"
                  >
                    <svg
                      class="w-12 h-12"
                    >
                      changenow-logo.svg
                    </svg>
                  </td>
                  <td>
                    <div
                      class="font-semibold text-theme-primary-500 hover:text-theme-primary-400"
                    >
                      ChangeNOW Plugin
                    </div>
                    <div
                      class="inline-flex items-center space-x-2"
                    >
                      <span
                        class="text-theme-neutral-dark"
                      >
                        ChangeNOW
                      </span>
                    </div>
                  </td>
                  <td
                    class="py-10 text-center text-theme-neutral-dark"
                  >
                    Other
                  </td>
                  <td
                    class="flex justify-center py-10 text-theme-neutral-dark"
                  >
                    <div
                      class="flex"
                      data-testid="ReviewRating"
                    >
                      <div
                        class="relative my-auto"
                      >
                        <div
                          class="absolute overflow-hidden"
                          style="width: 95%;"
                        >
                          <svg
                            class="text-theme-warning-400 w-3"
                          >
                            star.svg
                          </svg>
                        </div>
                        <svg
                          class="text-theme-neutral w-3"
                        >
                          star.svg
                        </svg>
                      </div>
                      <span
                        class="leading-tight my-auto ml-1 {{ $ratingClass ?? '' }}"
                      >
                        4.8
                      </span>
                    </div>
                  </td>
                  <td
                    class="py-10 text-center text-theme-neutral-dark"
                  >
                    v 1.3.8
                  </td>
                  <td
                    class="w-16"
                  >
                    <button
                      class="sc-AxjAm ddJBbC"
                      color="primary"
                      type="button"
                    >
                      Install
                    </button>
                  </td>
                </tr>
                <tr
                  class="border-b border-dashed border-theme-neutral-200"
                >
                  <td
                    class="w-16"
                  >
                    <svg
                      class="w-12 h-12"
                    >
                      changenow-logo.svg
                    </svg>
                  </td>
                  <td>
                    <div
                      class="font-semibold text-theme-primary-500 hover:text-theme-primary-400"
                    >
                      Bold Ninja
                    </div>
                    <div
                      class="inline-flex items-center space-x-2"
                    >
                      <span
                        class="text-theme-neutral-dark"
                      >
                        Delegate Fun
                      </span>
                      <div
                        class="sc-AxirZ hPsuTj"
                        height="16"
                        width="16"
                      >
                        <svg>
                          grant.svg
                        </svg>
                      </div>
                    </div>
                  </td>
                  <td
                    class="py-10 text-center text-theme-neutral-dark"
                  >
                    Game
                  </td>
                  <td
                    class="flex justify-center py-10 text-theme-neutral-dark"
                  >
                    <div
                      class="flex"
                      data-testid="ReviewRating"
                    >
                      <div
                        class="relative my-auto"
                      >
                        <div
                          class="absolute overflow-hidden"
                          style="width: 97.50000000000001%;"
                        >
                          <svg
                            class="text-theme-warning-400 w-3"
                          >
                            star.svg
                          </svg>
                        </div>
                        <svg
                          class="text-theme-neutral w-3"
                        >
                          star.svg
                        </svg>
                      </div>
                      <span
                        class="leading-tight my-auto ml-1 {{ $ratingClass ?? '' }}"
                      >
                        4.9
                      </span>
                    </div>
                  </td>
                  <td
                    class="py-10 text-center text-theme-neutral-dark"
                  >
                    v 2.0.0
                  </td>
                  <td
                    class="w-16"
                  >
                    <button
                      class="sc-AxjAm ddJBbC"
                      color="primary"
                      type="button"
                    >
                      Install
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
